### PR TITLE
[kube-state-metrics] Ensure consistent telemetry configuration

### PR DESCRIPTION
**Pull request by treydock**
#### What this PR does / why we need it:

Ensure proper configuration of telemetry port and ensure port is exposed from the container.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

@tariq1890 @mrueg

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by treydock** [comment]
Realized didn't need to use `telemetryPort` for the actual deployment argument and container port since only needed for the service, so this pull request is really just to ensure the container exposes port 8081 that can then be used by the service.

**Review by mrueg — APPROVED** [review]

File: charts/kube-state-metrics/templates/deployment.yaml [hunk 1]
```diff
@@ -176,6 +176,9 @@ spec:
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         ports:
         - containerPort: 8080
+{{ if .Values.selfMonitor.enabled -}}
+        - containerPort: 8081
+{{ end -}}
```
Comment (scottrigby) [inline charts/kube-state-metrics/templates/deployment.yaml:181]: Re https://github.com/prometheus-community/helm-charts/pull/903/checks?check_run_id=2452191125

>Error:  templates/deployment.yaml: unable to parse YAML: error converting YAML to JSON: yaml: line 129: did not find expected key

At a quick glance, the trailing dash may be pulling the YAML up to the next line, making the deploy fail?

```suggestion
{{- if .Values.selfMonitor.enabled }}
        - containerPort: 8081
{{- end }}
```

Comment (treydock) [reply charts/kube-state-metrics/templates/deployment.yaml]: Yep, I just caught that rendering the templates locally, and made this change.

**Comment by treydock** [comment]
@scottrigby or @mrueg I think I need a re-review after latest fixes to address changes requested by @scottrigby

**Review by mrueg — APPROVED** [review]

**Pull request merged**